### PR TITLE
[SPARK-49438][SQL] Fix the pretty name of the `FromAvro` & `ToAvro` expression

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
In the PR, I propose to override the `prettyName` method of the `FromAvro` & `ToAvro` expression and set to `from_avro` & `to_avro` by default as in `FunctionRegistry`:
https://github.com/apache/spark/blob/6d8235f3b2bbaa88b10c35d6eecddffa4d1b04a4/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala#L873-L874

### Why are the changes needed?
To don't confuse users by non-existent function name, and print correct name in errors.

### Does this PR introduce _any_ user-facing change?
Yes.

### How was this patch tested?
Update existed UT.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: connector/avro/src/test/scala/org/apache/spark/sql/avro/AvroFunctionsSuite.scala [hunk 1]
```diff
@@ -328,7 +328,7 @@ class AvroFunctionsSuite extends QueryTest with SharedSparkSession {
              |select to_avro(s, 42) as result from t
              |""".stripMargin)),
         errorClass = "DATATYPE_MISMATCH.TYPE_CHECK_FAILURE_WITH_HINT",
-        parameters = Map("sqlExpr" -> "\"toavro(s, 42)\"",
```
Comment (panbingkun) [inline connector/avro/src/test/scala/org/apache/spark/sql/avro/AvroFunctionsSuite.scala:331]: Displaying `toavro` in the error message looks weird.

**Comment by panbingkun** [comment]
cc @MaxGekk @cloud-fan

**Review by hvanhovell — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Review by MaxGekk — APPROVED** [review]
Waiting for CI.

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @panbingkun and @HyukjinKwon @hvanhovell for review.

**Pull request closed without merging**